dev/release/verify-release-candidate.sh

```diff
@@ -404,62 +636,102 @@ test_python() {
   if [ "${ARROW_GANDIVA}" = "ON" ]; then
     export PYARROW_WITH_GANDIVA=1
   fi
+  if [ "${ARROW_PLASMA}" = "ON" ]; then
+    export PYARROW_WITH_PLASMA=1
+  fi
+  if [ "${ARROW_S3}" = "ON" ]; then
+    export PYARROW_WITH_S3=1
+  fi
+
+  pushd python
 
+  # Build pyarrow
   python setup.py build_ext --inplace
-  pytest pyarrow -v --pdb
+
+  # Check mandatory and optional imports
+  python -c "
+import pyarrow
+import pyarrow._hdfs
+import pyarrow.csv
+import pyarrow.dataset
+import pyarrow.fs
+import pyarrow.json
+import pyarrow.orc
+import pyarrow.parquet
+"
+  if [ "${ARROW_CUDA}" == "ON" ]; then
+    python -c "import pyarrow.cuda"
+  fi
+  if [ "${ARROW_FLIGHT}" == "ON" ]; then
+    python -c "import pyarrow.flight"
+  fi
+  if [ "${ARROW_GANDIVA}" == "ON" ]; then
+    python -c "import pyarrow.gandiva"
+  fi
+  if [ "${ARROW_PLASMA}" == "ON" ]; then
+    python -c "import pyarrow.plasma"
+  fi
+  if [ "${ARROW_S3}" == "ON" ]; then
+    python -c "import pyarrow._s3fs"
+  fi
+
+  # Install test dependencies
+  pip install -r requirements-test.txt
+
+  # Execute pyarrow unittests
+  pytest pyarrow -v
 
   popd
 }
 
 test_glib() {
-  pushd c_glib
+  show_header "Build and test C Glib libraries"
 
-  pip install meson
-
-  meson build --prefix=$ARROW_HOME --libdir=lib
-  ninja -C build
-  ninja -C build install
-
-  export GI_TYPELIB_PATH=$ARROW_HOME/lib/girepository-1.0:$GI_TYPELIB_PATH
+  # Build and test C GLib
+  setup_conda glib gobject-introspection meson ninja ruby || exit 1
+  setup_virtualenv meson || exit 1
 
+  # Install bundler if doesn't exist
   if ! bundle --version; then
     gem install --no-document bundler
   fi
 
+  local build_dir=$ARROW_TMPDIR/c-glib-build
+  mkdir -p $build_dir
+
+  pushd c_glib
+
+  # Build the C GLib bindings
+  meson \
+    --buildtype=${CMAKE_BUILD_TYPE:-release} \
+    --libdir=lib \
+    --prefix=$ARROW_HOME \
+    $build_dir
+  ninja -C $build_dir
+  ninja -C $build_dir install
+
+  # Test the C GLib bindings
+  export GI_TYPELIB_PATH=$ARROW_HOME/lib/girepository-1.0:$GI_TYPELIB_PATH
   bundle config set --local path 'vendor/bundle'
   bundle install
   bundle exec ruby test/run-test.rb
 
   popd
 }
 
-test_js() {
-  pushd js
-
-  if [ "${INSTALL_NODE}" -gt 0 ]; then
-    export NVM_DIR="`pwd`/.nvm"
-    mkdir -p $NVM_DIR
-    curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.0/install.sh | \
-      PROFILE=/dev/null bash
-    [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
+test_ruby() {
+  show_header "Build and test Ruby libraries"
 
-    nvm install --lts
-    npm install -g yarn
-  fi
+  # required dependencies are installed by test_glib
+  setup_conda || exit 1
+  setup_virtualenv || exit 1
```

Comment:
Please don't call them if we use system libraries.

BTW, I think that we can omit `|| exit 1` because we use `set -e`.

Reply:
`|| exit 1` is there to handle the return codes from the functions like `return 1` which is used to early terminate without actually stopping the script.